src/plugins/intel_npu/src/plugin/npuw/partitioning/partitioning.cpp

```diff
@@ -2002,6 +2003,34 @@ void Partitioner::optimize(const std::string& func_name) {
             });
         }
 
+        // Gather NF4 closures in compile time, where requested
+        for (auto&& p : ctx.params_to_nf4_gather) {
+            const auto& tensor_to_gather = p.second;
+            auto w_idx = f._model->get_parameter_index(tensor_to_gather.w);
+
+            // Need to add a new parameter right away, since it's going to be processed by the unpack below
+            f._model->add_parameters({p.first});
+            to_remove.push_back(tensor_to_gather.w);
+            to_remove_idx.insert(w_idx);
+
+            ov::npuw::util::non_parallel_for(func_group.refs.size(), [&](std::size_t f_idx) {
+                auto& funcall = func_group.refs[f_idx].get();
```

Comment:
just a plain range-based for loop works here - as you don't really need `f_idx`?

Reply:
We use `ov::npuw::util::non_parallel_for(func_group.refs.size(), [&](std::size_t f_idx) {` for all other transformations in `optimize()` - unpack, concat, permute, etc. I'd keep it consistent